File: 10-projects/todo-app/TodoModel.swift
import SwiftData
import Foundation

@Model
final class TodoItem {
    var title: String
    var isDone: Bool
    var createdAt: Date
    var priority: Priority

    enum Priority: Int, Codable, CaseIterable {
        case low = 0, medium = 1, high = 2
        var label: String { ["Low", "Medium", "High"][rawValue] }
        var color: String { ["blue", "orange", "red"][rawValue] }
    }

    init(title: String, priority: Priority = .medium) {
        self.title = title
        self.isDone = false
        self.createdAt = Date()
        self.priority = priority
    }
}

File: 10-projects/todo-app/flutter_equivalent/todo_model.dart
enum Priority { low, medium, high }

extension PriorityLabel on Priority {
  String get label => name[0].toUpperCase() + name.substring(1);
}

class TodoItem {
  final String id;
  String title;
  bool isDone;
  Priority priority;
  final DateTime createdAt;

  TodoItem({
    required this.title,
    this.priority = Priority.medium,
  })  : id = DateTime.now().toIso8601String(),
        isDone = false,
        createdAt = DateTime.now();
}
